### PR TITLE
Fix sign-compare warnings issued by AppleClang

### PR DESCRIPTION
**Pull request by kunaltyagi**

### CODE REVIEW

File: features/include/pcl/features/impl/fpfh.hpp [hunk 1]
```diff
@@ -118,6 +120,7 @@ pcl::FPFHEstimation<PointInT, PointNT, PointOutT>::weightPointSPFHSignature (
   const auto nr_bins_f2 = hist_f2.cols ();
   const auto nr_bins_f3 = hist_f3.cols ();
   const auto nr_bins_f12 = nr_bins_f1 + nr_bins_f2;
+  using iteration_type = std::remove_cv_t<decltype(nr_bins_f1)>;
```
Comment (SergioRAgostinho) [inline features/include/pcl/features/impl/fpfh.hpp]: I would prefer something along these lines
```cpp 
using iteration_type = decltype(hist_f1)::Index;
```

**Comment by kunaltyagi** [comment]
CI is green (except 19.10)

File: features/include/pcl/features/impl/fpfh.hpp (hunk 1 above)
Comment (kunaltyagi) [reply features/include/pcl/features/impl/fpfh.hpp]: I directly used `Index`

File: features/include/pcl/features/impl/fpfh.hpp [hunk 2]
```diff
@@ -44,6 +44,8 @@
 #include <pcl/features/fpfh.h>
 #include <pcl/features/pfh_tools.h>
 
+#include <type_traits>
+
```
Comment (SergioRAgostinho) [inline features/include/pcl/features/impl/fpfh.hpp]: You no longer need this here.

Comment (kunaltyagi) [reply features/include/pcl/features/impl/fpfh.hpp]: Removed

**Comment by kunaltyagi** [comment]
Squash the last 3 commits?

**Pull request merged**